### DISCUSSION TITLE
pkg upgrade fails after install openssl

### DISCUSSION THREAD

**Discussion by f-zl** [termux/termux-packages#12034] [Q&A]
When I run nginx, it says libssl.so.3 is not found. So I run `pkg install openssl` and install openssl/stable 3.0.5. But then nginx says libcrypto.so.1.1 is not found. Also `pkg upgrade` fails for not finding libssl.so.1.1. How can I make it work?

**Comment by sylirre** [comment]
As Termux is a rolling distribution, everything should be upgraded before installing a new package. Debian package management workflow doesn't apply there, even though we use `apt` as package manager. This is described at https://wiki.termux.com/wiki/Package_Management.

Your Termux environment is broken now because `openssl` was upgraded to a new major version without upgrading package manager and other dependent things. Now you can just reset your installation and start with a new one:
```
cp -a $PREFIX $HOME/usr.bak
rm -rf $PREFIX
```
Then restart application and do `yes | pkg upgrade`. `usr.bak` may be deleted once you copied your configuration files from it.
